### PR TITLE
RISC-V SoC folder cleanup

### PR DESCRIPTION
**Pull request by gmarull**
This PR is an attempt to partially clean up the mess in soc/riscv/, specially the riscv-privileged folder. In summary, this PR does:

- Drops the "riscv-privileged" SoC family, as it didn't make any sense. RISC-V privileged is a spec, not a SoC family.
- Each SoC/family previously defined under soc/riscv-privileged is now moved to soc/riscv. Single SoCs that artificially created series to adapt to the SoC structure requirements have been simplified to just define a single SoC.
- Some SoC/families had incorrect naming (not even following vendor specs), this has been fixed but may contain issues, please see the commits for more details.

Further enhancements will come after this PR gets merged.

### CODE REVIEW

**Comment by carlocaione** [comment]
I'm happy with this if CI is happy with this.

**Review by npitre — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by nashif — APPROVED** [review]

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -3384,7 +3384,7 @@ ITE Platforms:
     - drivers/*/*it8xxx2*.c
     - dts/bindings/*/*ite*
```
Comment (kartben) [inline MAINTAINERS.yml:3385]: While you're at it could you make this one more specific if possible, please? It currently matches "display/ilitek..."

**Review by carlocaione — APPROVED** [review]

File: MAINTAINERS.yml (hunk 1 above)
Comment (gmarull) [reply MAINTAINERS.yml:3385]: I'll open a follow-up PR with more risc-v enhancements

**Comment by tejlmand** [comment]
A little late, but lgtm 👍

**Pull request merged**